kore/src/Kore/Step/Rule/Simplify.hs

```diff
@@ -79,10 +83,12 @@ simplifyRuleLhs rule@(RulePattern _ _ _ _ _ _) = do
         Conditional.withCondition
             left
             $ Predicate.fromPredicate
-                ( makeAndPredicate
-                    requires
-                    (makeCeilPredicate left)
-                )
+                lhsPredicate
+    lhsPredicate =
```

Comment:
Could we move `lhsPredicate` and `definedLhs` into a `let` block above? I think this bit would be easier to understand if they were closer to where they are used.